### PR TITLE
Pressing Backspace in an empty cell deletes the cell

### PR DESCRIPTION
**Pull request by agbaum**
## References

#7713 

## Code changes

Added a new signal in the code mirror which is emitted when backspace is pressed while the editor is empty.

Added a function in the notebook widget to handle the signal and delete the active cell.

Added three tests:

- Check that the signal is released when backspace is pressed in an empty editor.
- Check that the signal is *not* released when backspace is pressed in a non-empty editor. 
- Check that the notebook deletes the active cell when it receives the signal.

## User-facing changes

Pressing backspace in an empty cell now deletes the cell. 
Some discussion was had around whitespace, but for now I decided to just delete completely empty cells, since we hadn't come to a conclusion and the implementation was easier.

### CODE REVIEW

**Comment by blink1073** [comment]
Thanks @agbaum!  Because we're in the 2.0 release cycle and this affects an API, I marked it for 2.1.

**Comment by blink1073** [comment]
Hi @agbaum, we're prepping for a release this week.   Are you available to help finish this PR?  If so, can you please run `jlpm run integrity` and push those changes?

File: packages/notebook/src/widget.ts [hunk 1]
```diff
@@ -1536,6 +1537,22 @@ export class Notebook extends StaticNotebook {
     this.mode = 'edit';
   }
 
+  /**
+   * Handle emtpy backspace signals from cells (delete the cell).
```
Comment (jasongrout) [inline packages/notebook/src/widget.ts]:
```suggestion
   * Handle empty backspace signals from cells (delete the cell).
```

**Comment by jasongrout** [comment]
Looking at this: we already have a command to restart and run all (it's available from the menu). Is there a reason why this new button implements the logic itself instead of just firing off the command? You could even use a CommandButton to do most of the logic for you.

Also, CC @tgeorgeux for UX review for the icon. Note that this icon is what is used in the classic notebook these days for this action: 
Screen Shot 2020-03-25 at 8 05 19 AM
I personally like the restart and run all icon from the nteract discussion, though (perhaps with the arrow going the other way, though): https://github.com/nteract/nteract/issues/4809#issuecomment-575374186

>  If so, can you please run `jlpm run integrity` and push those changes?

@blink1073 - interestingly, on this branch, I run `jlpm run integrity` and it doesn't show any problems.

**Comment by jasongrout** [comment]
> If so, can you please run `jlpm run integrity` and push those changes?

Weird - I'm not seeing integrity issues locally, and I can't seem to restart the CI to test again, and the CI build that failed seems to be no longer available.

File: packages/notebook/src/widget.ts [hunk 2]
```diff
@@ -1536,6 +1537,22 @@ export class Notebook extends StaticNotebook {
     this.mode = 'edit';
   }
 
+  /**
+   * Handle empty backspace signals from cells (delete the cell).
+   */
+
+  private _onEmptyBackspace(editor: CodeEditor.IEditor) {
+    let activeCellIndex = this.activeCellIndex;
+    const model = this.model!;
```
Comment (saulshanabrook) [inline packages/notebook/src/widget.ts]: Can we check 
```suggestion
    const model = this.model;
    if (!model) {
      return;
    }
```

**Comment by saulshanabrook** [comment]
Works for me!

**Pull request merged**